records/2024-02-15.private-eks-cluster.md

```diff
@@ -0,0 +1,85 @@
+# Converting To A Private EKS Cluster
+
+Date: 2024-02-15
+
+## Status
+
+**DRAFT**.
+
+## Context
+
+GC Notify hosts a large portion of its system in an AWS managed Elastic Kubernetes Service. When deploying an EKS cluster, there is the option to make the main administrative endpoint either public or private. Public endpoints are available to the entirety of the internet, but protected by the AWS IAM and Single-Sign-On (SSO). Private endpoints are also protected by AWS IAM and SSO but are also not accessible via the open internet. 
```

Comment:
Which github actions would we run from the internally running github instances (ARC)?

Reply:
We would only run anything touching kubectl/helmfile on the internal actions. In cases where the kubernetes deploy is integrated with the docker build action, I have extracted the kubernetes section into its own action, and the original now triggers this remote action. This way we can reduce the load on our k8s cluster, and just run the necessary steps internally. It also allows us to consolidate the K8s actions into a single repository (manifests). I've added a bit about [consolidating the k8s actions here.](https://github.com/cds-snc/notification-adr/pull/61/files#diff-b43c07e17cc223e9ec3ad3dd4ecca0862e8b6f39427f96469f35a9b4385a23cfR65)